### ISSUE TITLE
ofAVFoundationPlayer doesn't support OF_LOOP_PALINDROME

### ISSUE THREAD

**Issue by mattfelsen** [openframeworks/openFrameworks#3548]
Only supports normal looping and logs a warning if you try to set palindrome looping:

https://github.com/openframeworks/openFrameworks/blob/5c463da6781a939dc595360a50965152dd42d50e/libs/openFrameworks/video/ofAVFoundationPlayer.mm#L325-L327

Seems like a core feature to not support...should this be milestoned to 0.9? I don't see any team names so I'll mention @prisonerjohn @obviousjim :)

**Comment by kylemcdonald** [comment]
good call, this is an old feature and while it's an obvious bug and regression i'm not sure i'd call it critical, so i'll milestone for 0.10.0

if this can't be added relatively easily, would it be reasonable to remove this as a feature? or is it better to just not support it on OS X?

@openframeworks/video is the team ;)

**Comment by julapy** [comment]
playing video backwards can work with AVFoundation using the rate property,
https://developer.apple.com/library/ios/documentation/AVFoundation/Reference/AVPlayer_Class/index.html#//apple_ref/occ/instp/AVPlayer/rate
although the `canPlayReverse` property needs to be checked first.
it comes down to the video format being used and how the video is encoded which determines if it can be played backwards.

lets keep this issue open as it can be addressed.

**Comment by i-n-g-o** [comment]
this issue is related to this one: https://github.com/openframeworks/openFrameworks/issues/3629
the current implementation of ofAVFoundationVideoPlayer does not allow backwards-playback.
as soon as the backwards-issue is fixed, playing videos in palidrome would be possible.

**Comment by DomAmato** [comment]
I don't think this is supported in WMF as playing in reverse doesn't seem well supported in the newest media handling from windows, its an ugly affair and requires thinning the frames depending on the codec which leads to choppy playback. This isn't an issue currently as the video player doesn't utilize WMF rather DirectShow but in going forward if it is upgraded this will be a problem.

**Comment by ofTheo** [comment]
I would be in favor of deprecating / removing OF_LOOP_PALINDROME 
It was really just a side effect of wrapping the Quicktime 7 loop modes which isn't used on Windows or OS X. 

I think Linux GST still supports OF_LOOP_PALINDROME by reversing the speed when the movie reaches the end. Not sure how well it works across video types etc. 

https://github.com/openframeworks/openFrameworks/blob/5c463da6781a939dc595360a50965152dd42d50e/libs/openFrameworks/video/ofGstUtils.cpp#L636

**Comment by danoli3** [comment]
This has been a long standing missing lack of feature... push to 0.10.0? ;)

**Comment by arturoc** [comment]
yes, i moved it to 0.10 and we can probably just deprecate it then
